server/src/main/java/uk/ac/cam/cl/echo/extrusionfinder/server/parts/Size.java

```diff
@@ -0,0 +1,100 @@
+package uk.ac.cam.cl.echo.extrusionfinder.server.parts;
+
+public class Size {
+
+    public enum Unit {
+        IN, // inches
+        MM, // millimeters
+        UNKNOWN
+    }
+
+    /**
+     * Dimension1 is either width/diameter (if one measurement), width (if two),
+     * or outer diameter (if tube).
+     */
+    private float dimension1;
+
+    /**
+     * Dimension2 is either 0 (if one measurement), height (if two), or inner
+     * diameter (if tube).
+     */
+    private float dimension2;
+
+    /**
+     * The unit of the measurements.
+     */
+    private Unit unit;
+
+    /**
+     * Default empty constructor. Measurements are set to 0, unit is unknown.
+     */
+    public Size() {
+        dimension1 = 0;
+        dimension2 = 0;
```

Comment:
`0` seems rather close to a valid value; I'm not sure it's sensible to be using it as a null state. Why aren't you just using `null` instead of empty `Size` objects though?